### PR TITLE
Add deprecation issue template

### PR DESCRIPTION
**Pull request by YuKitsune**
Adds an issue template for deprecations.

[sc-51549]

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/deprecation.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+name: Deprecation
+about: For the Octopus team to make customers aware of upcoming deprecations.
+title: ''
+labels: tag/breaking-change
```
Comment (tekerson) [inline .github/ISSUE_TEMPLATE/deprecation.md:5]: I don't think we want to default to breaking change. As discussed this morning, adding the deprecation shouldn't effect behaviour at all.

I can't really see any labels that we *do* want though; maybe we should have a `kind/deprecation` label?

File: .github/ISSUE_TEMPLATE/deprecation.md
Comment (adam-mccoy) [inline .github/ISSUE_TEMPLATE/deprecation.md:1]: I think it'd be worth adding the Process section from Brenton's PR [here](https://github.com/OctopusDeploy/Issues/issues/8192) as part of the template. It outlines the schedule for deprecations which would be useful for customers.

File: .github/ISSUE_TEMPLATE/deprecation.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+assignees: ''
+
+---
+
+**_Are you a customer of Octopus Deploy? Please contact [our support team](https://octopus.com/support) if you have any questions regarding an upcoming deprecation.**
+
+# Prerequisites
+
+- [ ] I have updated the [public deprecation schedule](https://octopus.com/docs/deprecations)
```
Comment (tekerson) [inline .github/ISSUE_TEMPLATE/deprecation.md:14]: I like the idea of having this as a checkbox *somewhere*, but I don't think that document would be updated at the time this issue is created (assuming we're creating it at the same time as the internal RFC)

File: .github/ISSUE_TEMPLATE/deprecation.md (hunk 1 above)
Comment (YuKitsune) [reply .github/ISSUE_TEMPLATE/deprecation.md:5]: Could also just not have any labels at all. This is primarily for release notes after all.

File: .github/ISSUE_TEMPLATE/deprecation.md
Comment (YuKitsune) [reply .github/ISSUE_TEMPLATE/deprecation.md:1]: I like it! I've brought it over, but made it a bit more generic.

File: .github/ISSUE_TEMPLATE/deprecation.md (hunk 1 above)
Comment (tekerson) [reply .github/ISSUE_TEMPLATE/deprecation.md]: Yeh, I was just generally asking the question. I don't know if having a label is important for anyone. I don't think it is for us, so I'd be happy not to have any in the template.

File: .github/ISSUE_TEMPLATE/deprecation.md (hunk 2 above)
Comment (YuKitsune) [reply .github/ISSUE_TEMPLATE/deprecation.md:10]: I don't think it makes a huge difference which one comes first, as long as the public notice is up by the time the issue is closed. No strong opinions here, so open to moving it somewhere else.

**Review by tekerson — APPROVED** [review]

**Pull request merged**